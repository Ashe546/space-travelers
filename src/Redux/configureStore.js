import { configureStore } from '@reduxjs/toolkit';

import rocketReducer from './rocket/rocket';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
  },
});

export default store;
