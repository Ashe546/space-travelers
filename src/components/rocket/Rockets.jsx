import { useSelector, useDispatch } from 'react-redux';

import { useEffect } from 'react';
import { fetchRocket } from '../../Redux/rocket/rocket';

import Rocket from './Rocket';

function Rockets() {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchRocket());
  }, [dispatch]);

  const rockets = useSelector((state) => state.rockets);

  return (
    <main>
      {rockets.map((rocket) => (
        <Rocket
          key={rocket.id}
          id={rocket.id}
          description={rocket.description}
          title={rocket.name}
          image={rocket.image}
        />
      ))}
    </main>
  );
}

export default Rockets;
