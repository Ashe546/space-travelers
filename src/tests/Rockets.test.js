import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import Rockets from '../components/rocket/Rockets';

const mockStore = configureStore([]);

describe('Rockets', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      rockets: [
        {
          id: 1,
          name: 'Falcon 1',
          description:
            'Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009',
          image: 'https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png',
          reservation: false,
        },
        {
          id: 2,
          name: 'Falcon 9',
          description:
            'Falcon 9 is a two-stage rocket designed and manufactured by SpaceX for the reliable and safe transport of satellites and the Dragon spacecraft into orbit.',
          image: 'https://images2.imgbox.com/4f/e3/I0lkuJ2e_o.png',
          reservation: false,
        },
        {
          id: 3,
          name: 'Falcon Heavy',
          description:
            'Falcon Heavy is a heavy lift launch vehicle designed and manufactured by SpaceX for the reliable and safe transport of satellites, the Dragon spacecraft, and other payloads into orbit.',
          image: 'https://images2.imgbox.com/3d/86/cnu0pan8_o.png',
          reservation: false,
        },
        {
          id: 4,
          name: 'Starship',
          description:
            'Starship is an upcoming super heavy-lift launch vehicle and spacecraft under development by SpaceX.',
          image: 'https://images2.imgbox.com/be/e7/iNqsqVYM_o.png',
          reservation: false,
        },
      ],
    });

    component = renderer.create(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
  });

  it('should render rocket components correctly with given state from Redux', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
