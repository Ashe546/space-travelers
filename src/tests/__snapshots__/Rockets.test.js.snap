// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets renders correctly 1`] = `
<main>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Falcon 1
        </h2>
        <p>
          Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/4f/e3/I0lkuJ2e_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Falcon 9
        </h2>
        <p>
          Falcon 9 is a two-stage rocket designed and manufactured by SpaceX for the reliable and safe transport of satellites and the Dragon spacecraft into orbit.
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/3d/86/cnu0pan8_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Falcon Heavy
        </h2>
        <p>
          Falcon Heavy is a heavy lift launch vehicle designed and manufactured by SpaceX for the reliable and safe transport of satellites, the Dragon spacecraft, and other payloads into orbit.
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/be/e7/iNqsqVYM_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Starship
        </h2>
        <p>
          Starship is an upcoming super heavy-lift launch vehicle and spacecraft under development by SpaceX.
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
</main>
`;

exports[`Rockets renders rocke correctly 1`] = `
<main>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Falcon 1
        </h2>
        <p>
          Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/4f/e3/I0lkuJ2e_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Falcon 9
        </h2>
        <p>
          Falcon 9 is a two-stage rocket designed and manufactured by SpaceX for the reliable and safe transport of satellites and the Dragon spacecraft into orbit.
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/3d/86/cnu0pan8_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Falcon Heavy
        </h2>
        <p>
          Falcon Heavy is a heavy lift launch vehicle designed and manufactured by SpaceX for the reliable and safe transport of satellites, the Dragon spacecraft, and other payloads into orbit.
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/be/e7/iNqsqVYM_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Starship
        </h2>
        <p>
          Starship is an upcoming super heavy-lift launch vehicle and spacecraft under development by SpaceX.
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
</main>
`;

exports[`Rockets renders rocket components correctly 1`] = `
<main>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Falcon 1
        </h2>
        <p>
          Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/4f/e3/I0lkuJ2e_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Falcon 9
        </h2>
        <p>
          Falcon 9 is a two-stage rocket designed and manufactured by SpaceX for the reliable and safe transport of satellites and the Dragon spacecraft into orbit.
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/3d/86/cnu0pan8_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Falcon Heavy
        </h2>
        <p>
          Falcon Heavy is a heavy lift launch vehicle designed and manufactured by SpaceX for the reliable and safe transport of satellites, the Dragon spacecraft, and other payloads into orbit.
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/be/e7/iNqsqVYM_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Starship
        </h2>
        <p>
          Starship is an upcoming super heavy-lift launch vehicle and spacecraft under development by SpaceX.
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
</main>
`;

exports[`Rockets should render rocket components correctly 1`] = `
<main>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Falcon 1
        </h2>
        <p>
          Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/4f/e3/I0lkuJ2e_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Falcon 9
        </h2>
        <p>
          Falcon 9 is a two-stage rocket designed and manufactured by SpaceX for the reliable and safe transport of satellites and the Dragon spacecraft into orbit.
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/3d/86/cnu0pan8_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Falcon Heavy
        </h2>
        <p>
          Falcon Heavy is a heavy lift launch vehicle designed and manufactured by SpaceX for the reliable and safe transport of satellites, the Dragon spacecraft, and other payloads into orbit.
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/be/e7/iNqsqVYM_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Starship
        </h2>
        <p>
          Starship is an upcoming super heavy-lift launch vehicle and spacecraft under development by SpaceX.
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
</main>
`;

exports[`Rockets should render rocket components correctly with given state fr 1`] = `
<main>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Falcon 1
        </h2>
        <p>
          Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/4f/e3/I0lkuJ2e_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Falcon 9
        </h2>
        <p>
          Falcon 9 is a two-stage rocket designed and manufactured by SpaceX for the reliable and safe transport of satellites and the Dragon spacecraft into orbit.
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/3d/86/cnu0pan8_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Falcon Heavy
        </h2>
        <p>
          Falcon Heavy is a heavy lift launch vehicle designed and manufactured by SpaceX for the reliable and safe transport of satellites, the Dragon spacecraft, and other payloads into orbit.
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/be/e7/iNqsqVYM_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Starship
        </h2>
        <p>
          Starship is an upcoming super heavy-lift launch vehicle and spacecraft under development by SpaceX.
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
</main>
`;

exports[`Rockets should render rocket components correctly with given state from Redux 1`] = `
<main>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Falcon 1
        </h2>
        <p>
          Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/4f/e3/I0lkuJ2e_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Falcon 9
        </h2>
        <p>
          Falcon 9 is a two-stage rocket designed and manufactured by SpaceX for the reliable and safe transport of satellites and the Dragon spacecraft into orbit.
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/3d/86/cnu0pan8_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Falcon Heavy
        </h2>
        <p>
          Falcon Heavy is a heavy lift launch vehicle designed and manufactured by SpaceX for the reliable and safe transport of satellites, the Dragon spacecraft, and other payloads into orbit.
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="rocket"
    >
      <img
        alt="spaceX"
        src="https://images2.imgbox.com/be/e7/iNqsqVYM_o.png"
      />
      <div
        className="details"
      >
        <h2>
          Starship
        </h2>
        <p>
          Starship is an upcoming super heavy-lift launch vehicle and spacecraft under development by SpaceX.
        </p>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
</main>
`;
